ddtrace/tracer/context.go

```diff
@@ -46,6 +48,13 @@ func StartSpanFromContext(ctx context.Context, operationName string, opts ...Sta
 	if s, ok := SpanFromContext(ctx); ok {
 		opts = append(opts, ChildOf(s.Context()))
 	}
+	opts = append(opts, WithContext(ctx))
 	s := StartSpan(operationName, opts...)
+	// If profiler labels were applied for this span, use the derived ctx that
+	// includes them. Otherwise a child of this span wouldn't be able to
+	// correctly restore the labels of its parent when it finishes.
+	if span, ok := s.(*span); ok && span.labelContext != nil {
+		ctx = span.labelContext
```

Comment:
Won't this cause losing everything that was in the original context? I think a user would expect that the returned context is equal to the original one (`ctx`) plus the `Span`.

Maybe we need to further enhance here?